### PR TITLE
[PyTorch/C] Fix compiling warnings and backend selection logic for fused attention

### PR DESCRIPTION
**Pull request by cyanguwa**
This PR fixes the compiling warnings in SDPA flash after upgrading to cuDNN frontend v1. It also fixes the logic for selecting the SDPA flash backend on sm80 when cuDNN runtime version is >8.9.6.

### CODE REVIEW

**Comment by cyanguwa** [comment]
Pipeline 11413511

**Comment by cyanguwa** [comment]
Pipeline 11414037

File: transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu [hunk 1]
```diff
@@ -224,11 +221,21 @@ void fused_attn_arbitrary_seqlen_fwd_impl(
                 std::make_tuple(nullptr), key_tensors_tuple,
                 Stats_tuple, bias_tuple, padding_tuple, dropout_tuple);
 
-            mha_graph->validate();
-            mha_graph->build_operation_graph(handle);
-            mha_graph->create_execution_plans({fe::HeurMode_t::A});
-            mha_graph->check_support(handle);
-            mha_graph->build_plans(handle);
+            if (!mha_graph->validate().is_good()) {
+                NVTE_ERROR("MHA Graph (fwd) validation is unsuccessful.");
```
Comment (ptrendx) [inline transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu:225]: Can we instead introduce something similar to https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/common/util/logging.h#L54?

Comment (cyanguwa) [reply transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu]: Done.

**Comment by cyanguwa** [comment]
Pipeline 11439646

**Comment by cyanguwa** [comment]
With newer cuDNN, pipeline 11441309

**Review by ptrendx — APPROVED** [review]

**Comment by cyanguwa** [comment]
Pipelines 11444120 (PyTorch and Paddle) and 11464931 (Jax with an older Jax container) passed.

**Pull request merged**